### DISCUSSION TITLE
One User Experiencing Continuous Token Refresh During Login

### DISCUSSION THREAD

**Discussion by wizzymotion** [supabase/supabase#21508] [Questions]
# Hi Supafriends! 🌟

I'm encountering a peculiar issue with just one user among the 1,000+ who joined since our last release.

## Issue Description 🚨
- **Problem**: When this user enters their login credentials, the authentication process enters an unusual state.
- **Symptom**: The AuthCode seems to malfunction, resulting in a continuous loop of token refreshes.

## Steps Taken 🛠️
- **Verification**: I've tested this user's credentials on my computer, and everything works fine.
- **Remote Testing**: Attempting to access the application from the user's computer remotely (using my credentials) also fails.

## Error Details 📝
- **Console Error**: The console shows an error related to `sameSite`, as illustrated in the attached image.

![kalebe_same_site_error](https://github.com/supabase/supabase/assets/78563491/66fd5df7-268e-4a84-9402-d1031d292710)

**Comment by bearbricknik** [comment]
Hi, I came across the same issue that the user runs into an infinite refresh of the token and each request returns 200 and actually a refreshed token. This continues until the user runs into a rate limit 429 and automatically log out the user.

Our supabase client actually uses custom sameSite and secure:

  const supabase = createClientComponentClient({
    cookieOptions: {
    name: process.env.NEXT_PUBLIC_COOKIE_NAME as string,
    sameSite: "none" as const, // "none" "strict"
    domain: getDomain(),
    path: "/",
    secure: true, // true false
  }
  })
  
  Where you able to fix this issue? I can not seem to find a way around it.

**Comment by krushmoto** [comment]
This happened to a few users of my app. They would attempt to login, then would enter an infinite refresh token loop until they hit the 429 error. Turned out to be the system clock on their windows device was off, basically invalidating all tokens/exchanges. After having them "Sync" the time on their windows machine the issue went away. Hope this helps!

**Reply by JellYouness** [reply]
thank you so much, for this answer, it helped a lot. i had a wrong time on my machine (-1 hour) and make supabase always think that the token is expired
